index.js

```diff
@@ -16,6 +16,7 @@ var through = require('through2')
 var util = require('util')
 
 var BASE_URI = 'https://www.googleapis.com/upload/storage/v1/b'
+var EXPIRED_UPLOAD_STATUS_CODE = 410
```

Comment:
Interesting, thanks for the clarification. After this PR, we will restart on a 404 and a 410, otherwise, we will pass along the API error to the user.